Makefile

```
@@ -14,15 +14,16 @@ CONTROLLER_DIR=${ROOT_DIR}/../${AWS_SERVICE}-controller
 CONTROLLER_BOOTSTRAP=./bin/controller-bootstrap
 CODE_GEN_DIR=${ROOT_DIR}/../code-generator
 
-AWS_SDK_GO_VERSION=$(shell curl -H "Accept: application/vnd.github.v3+json" \
-    https://api.github.com/repos/aws/aws-sdk-go/releases/latest | jq -r '.tag_name')
-ACK_RUNTIME_VERSION=$(shell curl -H "Accept: application/vnd.github.v3+json" \
-    https://api.github.com/repos/aws-controllers-k8s/runtime/releases/latest | jq -r '.tag_name')
-TEST_INFRA_COMMIT_SHA=$(shell curl -H "Accept: application/vnd.github.v3+json" \
-    https://api.github.com/repos/aws-controllers-k8s/test-infra/commits | jq -r ".[0].sha")
+ACK_RUNTIME_VERSION:=$(or $(shell echo $(ACK_RUNTIME_VERSION)),$(shell curl -H "Accept: application/vnd.github.v3+json" \
+    https://api.github.com/repos/aws-controllers-k8s/runtime/releases/latest | jq -r '.tag_name'))
+AWS_SDK_GO_VERSION:=$(or $(shell echo $(AWS_SDK_GO_VERSION)),$(shell curl -H "Accept: application/vnd.github.v3+json" \
+    https://api.github.com/repos/aws/aws-sdk-go/releases/latest | jq -r '.tag_name'))
+TEST_INFRA_COMMIT_SHA:=$(or $(shell echo $(TEST_INFRA_COMMIT_SHA)),$(shell curl -H "Accept: application/vnd.github.v3+json" \
+    https://api.github.com/repos/aws-controllers-k8s/test-infra/commits | jq -r ".[0].sha"))
```

Did you test that this works? What was your testing strategy?  I expect `echo $(ACK_RUNTIME_VERSION)` returning an empty string and setting that instead of shell command.

```suggestion
ACK_RUNTIME_VERSION:=$(or $(ACK_RUNTIME_VERSION),$(shell curl -H "Accept: application/vnd.github.v3+json" \
    https://api.github.com/repos/aws-controllers-k8s/runtime/releases/latest | jq -r '.tag_name'))
AWS_SDK_GO_VERSION:=$(or $(AWS_SDK_GO_VERSION),$(shell curl -H "Accept: application/vnd.github.v3+json" \
    https://api.github.com/repos/aws/aws-sdk-go/releases/latest | jq -r '.tag_name'))
TEST_INFRA_COMMIT_SHA:=$(or $(TEST_INFRA_COMMIT_SHA),$(shell curl -H "Accept: application/vnd.github.v3+json" \
    https://api.github.com/repos/aws-controllers-k8s/test-infra/commits | jq -r ".[0].sha"))
```

---

Thank you for pointing this out.   Initially, I have tested the approach without configuring the variables - I have noted the current, latest `aws-sdk-go` version (v1.44.46) and ACK runtime version (v0.19.2), in addition to the most recent ACK test-infra commit SHA (7378a3a23adbf93e014a9325cc6ef5209cbdf9da), were fetched and set in the `go.mod` and `requirements.txt` files respectively.

I have also tested with configuring the variables:  `export SERVICE=eks`, `export ACK_RUNTIME_VERSION=v0.19.1`, `export AWS_SDK_GO_VERSION=v1.44.42`, `export TEST_INFRA_COMMIT_SHA=fe6fadc3e1437f2785155c21e27dac3fd8c74f9b`, and then `make generate`. The configured values were set in the corresponding `go.mod` and `requirements.txt` files.

Also, I have tested the suggested changes (with/without variable configurations) with the desired results and updated the implementation.

---

Thank you, James!